### PR TITLE
Split tests so implementations can choose

### PR DESCRIPTION
**Pull request by carlossg**
which ones to call

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/workflow/ArtifactManagerTest.java [hunk 1]
```diff
@@ -117,52 +115,77 @@ public static void run(@Nonnull JenkinsRule r, @CheckForNull ArtifactManagerFact
...
+                assertFalse(b.getArtifactManager().delete());
+            }
+        });
+    }
+
+    /**
+     * @param image use {@link #prepareImage} in a {@link BeforeClass} block
+     */
+    public static void artifactStash(@Nonnull JenkinsRule r, @CheckForNull ArtifactManagerFactory factory, boolean weirdCharacters, @CheckForNull DockerImage image) throws Exception {
+        wrapInContainer(r, factory, weirdCharacters, new StashFunction(r, weirdCharacters, new TestStashFunction() {
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/ArtifactManagerTest.java:163]: Not a lambda?

**Review by jglick — COMMENTED** [review]
You need to file a PR against `master` (generally, an origin branch) in order for automatic deployment of binaries to run.

File: src/test/java/org/jenkinsci/plugins/workflow/ArtifactManagerTest.java [hunk 2]
```diff
@@ -117,52 +115,77 @@ public static void run(@Nonnull JenkinsRule r, @CheckForNull ArtifactManagerFact
...
+            if (runningContainer != null) {
+                runningContainer.close();
+            }
+        }
+    }
+
+    /**
+     * @param image use {@link #prepareImage} in a {@link BeforeClass} block
+     */
+    public static void artifactArchive(@Nonnull JenkinsRule r, @CheckForNull ArtifactManagerFactory factory, boolean weirdCharacters, @CheckForNull DockerImage image) throws Exception {
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/ArtifactManagerTest.java:129]: Javadoc should explain that unlike the `…AndDelete` variant, we assert that the manager ignores requests to delete things. Ditto for `artifactStash`.

**Comment by carlossg** [comment]
moved to https://github.com/jenkinsci/workflow-api-plugin/pull/71

**Comment by jglick** [comment]
Why reopened?

**Comment by carlossg** [comment]
it needs to be merged here or it won't build. Fails with the 2.89 build because I'm not a committer and my jenkinsfile changes are ignored

**Pull request closed without merging**